src/functions.rs

```diff
@@ -618,32 +622,44 @@ impl PortconRule<'_> {
 fn validate_port(
     port: &CascadeString,
     current_file: Option<&SimpleFile<String, String>>,
-) -> Result<(), ErrorItem> {
+) -> Result<Vec<Port>, ErrorItem> {
+    let mut ret = Vec::new();
     for substr in port.as_ref().split(',') {
-        if validate_port_helper(substr).is_err() {
-            return Err(ErrorItem::make_compile_or_internal_error(
-                "Not a valid port",
-                current_file,
-                port.get_range(),
-                "This should be a comma separated list of ports or port ranges",
-            ));
+        match validate_port_helper(substr) {
+            Ok(p) => ret.push(p),
+            Err(_) => {
+                return Err(ErrorItem::make_compile_or_internal_error(
+                    "Not a valid port",
+                    current_file,
+                    port.get_range(),
+                    "This should be a comma separated list of ports or port ranges",
+                ));
+            }
         }
     }
-    Ok(())
+    Ok(ret)
 }
 
-fn validate_port_helper(port: &str) -> Result<u16, ()> {
+fn validate_port_helper(port: &str) -> Result<Port, ()> {
     if port.contains('-') {
         let mut split = port.split('-');
         let first = split.next().ok_or(())?.parse::<u16>().map_err(|_| ())?;
         let second = split.next().ok_or(())?.parse::<u16>().map_err(|_| ())?;
         if split.next().is_some() || second <= first {
             Err(())
         } else {
-            Ok(0)
+            Ok(Port {
+                low_port_num: first,
+                high_port_num: Some(second),
+                range: None,
+            })
         }
     } else {
-        port.parse::<u16>().map_err(|_| ())
+        Ok(Port {
+            low_port_num: port.parse::<u16>().map_err(|_| ())?,
```

Comment:
We are dumping the error parse gives in validate_port above.  Is there any worthwhile info in that error that may be useful to pass to the user?  I am guessing it may to be too detailed.

Reply:
It gives messages like "invalid digit found in string", which seems a little to vague to be useful in my opinion.